spec/rspec/mocks/matchers/receive_spec.rb

```diff
@@ -267,6 +267,43 @@ def receiver.method_missing(*a); end # a poor man's stub...
         it_behaves_like "resets partial mocks cleanly" do
           let(:target) { allow(object) }
         end
+
+        context 'on a class method, from a class with subclasses' do
+          let(:superclass) { Class.new { def self.foo; "foo"; end }}
+          let(:subclass1)  { Class.new(superclass) { def self.foo; super.upcase; end }}
+          let(:subclass2)  { Class.new(superclass) }
+
+          context 'if the method is redefined in the subclass' do
+            it 'does not stub the method in the subclass' do
+              allow(superclass).to receive(:foo) { "foo!!" }
+              expect(superclass.foo).to eq "foo!!"
+              expect(subclass1.foo).not_to eq "foo!!"
+            end
+          end
+
+          context 'if the method is not redefined in the subclass' do
+            it 'stubs the method in the subclass' do
+              allow(superclass).to receive(:foo) { "foo!!" }
+              expect(superclass.foo).to eq "foo!!"
+              expect(subclass2.foo).to eq "foo!!"
+            end
+          end
+
+          it 'creates stub which can be called using `super` in a subclass' do
+            allow(superclass).to receive(:foo) { "foo!!" }
+            expect(subclass1.foo).to eq "FOO!!"
+          end
+
+          it 'can be used simultaneously in the superclass and subclasses' do
```

Comment:
What does "used simultaneously" mean here?  Do you mind "stubbed simultaneously"?